ethdb/database.go

```diff
@@ -68,20 +79,33 @@ type KeyValueStore interface {
 	io.Closer
 }
 
+var (
+	// ErrAncientUnknownKind is returned if the user attempts to read a kind that is
+	// not tracked by the AncientStore.
+	ErrAncientUnknownKind = errors.New("unknown kind")
+
+	// ErrAncientOutOrderInsertion is returned if the user attempts to inject out-of-order
+	// binary blobs into the AncientStore.
+	ErrAncientOutOrderInsertion = errors.New("the append operation is out-order")
+
+	// ErrAncientOutOfBounds is returned if the item requested is not contained within the
+	// AncientStore.
+	ErrAncientOutOfBounds = errors.New("out of bounds")
+)
+
 // AncientReader contains the methods required to read from immutable ancient data.
 type AncientReader interface {
-	// HasAncient returns an indicator whether the specified data exists in the
-	// ancient store.
-	HasAncient(kind string, number uint64) (bool, error)
```

Comment:
Any specific reason to remove this API?

Reply:
No, it just looked like dead code nobody cared about. I can put it back.

Reply:
Actually there was a reason: https://github.com/ethereum/go-ethereum/pull/23707#discussion_r735849401